File: lab/9-10/lab_9-10.py
def MultipleOfFive(arr: list[int]) -> int:
    for index, value in enumerate(arr):
        if value % 5 == 0:
            return index
    raise Exception(f'В массиве {arr} нет числа, кратного 5')


def ReplaceMaxWithZero(arr: list[int]) -> None:
    arr[arr.index(max(arr))] = 0


def DoubleElements(arr: list[int]) -> None:
    IMin = arr.index(min(arr))
    arr = arr[:IMin+1] + list(map(lambda x: x*2, arr[:IMin+1]))


def RowFill(row: list[int], value: int) -> None:
    row[:] = [value] * len(row)


def task_one() -> None:
    from random import randint
    n, a, b = map(int, input('Введите размер массивов и диапазон заполнения (от a до b): ').split())
    ArrOne = [randint(a, b) for i in range(n)]
    ArrTwo = [randint(a, b) for i in range(n)]
    
    print(f'Массив A: {ArrOne}\nМассив B: {ArrTwo}')
    
    if MultipleOfFive(ArrOne) < MultipleOfFive(ArrTwo):
        ReplaceMaxWithZero(ArrOne)
        DoubleElements(ArrTwo)
    elif MultipleOfFive(ArrOne) > MultipleOfFive(ArrTwo):
        ReplaceMaxWithZero(ArrTwo)
        DoubleElements(ArrOne)
    else:
        raise Exception('Индексы первых числел, кратных 5, совпали у обоих массивов')
     
    print(f'Массив A после изменений: {ArrOne}')
    print(f'Массив B после изменений: {ArrTwo}')


def task_two() -> None:
    from random import randint
    a, b = map(int, input('Введите диапазон заполнения (от a до b)').split())
    matrix = [[randint(a, b) for i in range(8)] for j in range(8)]
    print(f'Исходная матрица: {matrix}')

    for row in matrix:
        MaxValue = max(row)
        if row.count(MaxValue) == 1:
            RowFill(row, 1)
        else:
            RowFill(row, -1)
    print(f'Конечная матрица: {matrix}')

File: lab/9-10/lab_9-10.pas
function MultipleOfFive(Arr: array of integer): integer;
begin
  for var i := 0 to Arr.Length - 1 do
    if Arr[i] mod 5 = 0 then
    begin
      Result := i;
      exit
    end;
  raise new Exception('В массиве ' + Arr.ToString + ' нет числа, кратного 5');
end;


procedure ReplaceMaxWithZero(var Arr: array of integer);
begin
  Arr[Arr.IndexMax] := 0;
end;


procedure DoubleElements(var Arr: array of integer);
begin
  var IMin := Arr.IndexMin;
  Arr := Arr[:IMin+1] + Arr.Skip(IMin+1).Select(x -> x*2).ToArray;
end;


procedure RowFill(var mat: array[,] of integer; rowIndex: integer; val: integer);
begin
  for var j := 0 to mat.GetLength(1) - 1 do
    mat[rowIndex, j] := val;
end;


procedure task_one();
begin
  var (n, a, b) := readinteger3('Введите размер массивов и диапазон заполнения (от a до b): ');
  var ArrOne := ArrRandomInteger(n, a, b);
  var ArrTwo := ArrRandomInteger(n, a, b);
  
  Println('Массив A: ', ArrOne);
  Println('Массив B: ', ArrTwo);
  
  if MultipleOfFive(ArrOne) < MultipleOfFive(ArrTwo) then
  begin
    ReplaceMaxWithZero(ArrOne);
    DoubleElements(ArrTwo);
  end
  else if MultipleOfFive(ArrOne) > MultipleOfFive(ArrTwo) then
  begin
    ReplaceMaxWithZero(ArrTwo);
    DoubleElements(ArrOne);
  end
  else
    raise new Exception('Индексы первых числел, кратных 5, совпали у обоих массивов');

  Println('Массив A после изменений: ', ArrOne);
  Println('Массив B после изменений: ', ArrTwo);
end;


procedure task_two();
begin
  var (a, b) := ReadInteger2('Введите диапазон заполнения (от a до b)');
  var matrix := new integer[8,8];
  var MaxValue: integer;
  var row: array of integer;
  
  matrix.FillRandom(a, b);
  Println('Исходная матрица:', matrix);
  for var i := 0 to matrix.GetLength(0) - 1 do
  begin
    row := matrix.Row(i);
    MaxValue := row.Max;
    if row.Count(x -> x = MaxValue) = 1 then
      RowFill(matrix, i, 1)
    else
      RowFill(matrix, i, -1);
  end;
  Println('Конечная матрица', matrix);
end;